daemon/server/router/swarm/cluster_routes.go

```diff
@@ -171,6 +172,13 @@ func (sr *swarmRouter) getServices(ctx context.Context, w http.ResponseWriter, r
 		log.G(ctx).WithContext(ctx).WithError(err).Debug("Error getting services")
 		return err
 	}
+	if versions.LessThan(cliVersion, "1.25") {
+		legacyResponse := make([]*compat.Wrapper, 0, len(services))
+		for _, s := range services {
+			legacyResponse = append(legacyResponse, backFillLegacyNetwork(s))
+		}
+		return httputils.WriteJSON(w, http.StatusOK, legacyResponse)
+	}
```

Comment:
FWIW, I opened this PR before we disabled API < v1.44 by default, but probably won't do harm to keep these for now, and we can rip them out once we fully drop those API versions.